### PR TITLE
[Feat]: Threshold in dashboard for Value Component

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
PRD link: https://www.notion.so/signoz/PRD-Visualise-thresholds-in-dashboard-widgets-cde78a4773a14540a78de693086fd813?pvs=4

Close: https://github.com/SigNoz/signoz/issues/3913

https://www.loom.com/share/1ab5fa9d31f5419e9b965a2fe6220df3?sid=db495977-f2d9-4522-bd57-c36ea8fa8581

### CODE REVIEW

**Comment by palashgdev** [comment]
we can take this one later: 

while dropping highlight the area

**Comment by palashgdev** [comment]
user can't type in few dropdown in threshold area

**Comment by palashgdev** [comment]
dark and light mode does not have the parity

Screenshot 2023-11-13 at 6 21 16 PM

**Review by palashgdev — DISMISSED** [review]

File: frontend/src/components/ValueGraph/index.tsx [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
+			{isConflictingThresholds && (
+				<div
+					className={
+						threshold.thresholdFormat === 'Background'
+							? 'value-graph-bgconflict'
+							: 'value-graph-textconflict'
+					}
+				>
+					<Tooltip title="This value satisfies multiple thresholds">
+						<ExclamationCircleFilled style={{ color: '#E89A3C' }} />
```
Comment (YounixM) [inline frontend/src/components/ValueGraph/index.tsx:49]: can move style to scss file.

File: frontend/src/components/ValueGraph/index.tsx [hunk 2]
```diff
@@ -1,11 +1,63 @@
...
+			</Typography.Text>
+			{isConflictingThresholds && (
+				<div
+					className={
+						threshold.thresholdFormat === 'Background'
+							? 'value-graph-bgconflict'
+							: 'value-graph-textconflict'
+					}
+				>
+					<Tooltip title="This value satisfies multiple thresholds">
```
Comment (YounixM) [inline frontend/src/components/ValueGraph/index.tsx:48]: use translation

File: frontend/src/container/NewWidget/RightContainer/Threshold/ColorSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		}
+	}, [debounceColor, setColor]);
+
+	const handleColorChange = (_: Color, hex: string): void => {
+		setColorFromPicker(hex);
+	};
+
+	const items: MenuProps['items'] = [
+		{
+			key: '1',
```
Comment (YounixM) [inline frontend/src/container/NewWidget/RightContainer/Threshold/ColorSelector.tsx:32]: keys can be set to the value of color / `colorPicker`

File: frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+	const opacity = isDragging ? 0 : 1;
+	drag(drop(ref));
+
+	return (
+		<div
+			ref={ref}
+			style={{ opacity }}
+			data-handler-id={handlerId}
+			className="threahold-container"
```
Comment (YounixM) [inline frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx:157]: `threshold-container`

File: frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+		>
+			<Card
+				className={
+					isDarkMode
+						? `threahold-card threahold-card-dark`
```
Comment (YounixM) [inline frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx:162]: rename all instances of `threahold*` to `threshold`

File: frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+							className="threshold-action-icon"
+							onClick={deleteHandler}
+						/>
+					</div>
+					<div>
+						<Space>
+							<Typography.Text>If value is</Typography.Text>
+							{isEditMode ? (
+								<Select
+									style={{ minWidth: '73px', backgroundColor: '#141414' }}
```
Comment (YounixM) [inline frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx:184]: why do we specifically have the minWidth to `73px`?

**Review by YounixM — APPROVED** [review]
Minor comments, feel free to merge this and work on the review comments in subsequent PRs.

File: frontend/src/components/ValueGraph/index.tsx (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/components/ValueGraph/index.tsx:49]: Done.

File: frontend/src/components/ValueGraph/index.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply frontend/src/components/ValueGraph/index.tsx:48]: Done.

File: frontend/src/container/NewWidget/RightContainer/Threshold/ColorSelector.tsx (hunk 3 above)
Comment (Rajat-Dabade) [reply frontend/src/container/NewWidget/RightContainer/Threshold/ColorSelector.tsx:32]: This should be a hard-coded or read-only value, as the drop will get re-render as the key changes, I have update it with hard-coded color names.

File: frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx (hunk 5 above)
Comment (Rajat-Dabade) [reply frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx:162]: Done.

File: frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx (hunk 6 above)
Comment (Rajat-Dabade) [reply frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx:184]: Udpated it to max Width

File: frontend/src/container/NewWidget/RightContainer/constants.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+// > >= < <=  are the options
+export const operatorOptions: DefaultOptionType[] = [
+	{ value: '>', label: '>' },
+	{ value: '>=', label: '>=' },
+	{ value: '<', label: '<' },
+	{ value: '<=', label: '<=' },
+];
+
+// export const unitOptions = flattenedCategories.map((options) => ({
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply frontend/src/container/NewWidget/RightContainer/constants.ts:14]: Done.

**Review by palashgdev — APPROVED** [review]

**Comment by ankitnayan** [comment]
Closes https://github.com/SigNoz/signoz/issues/1600

**Pull request merged**